toolchain/parse/typed_nodes.h

```diff
@@ -677,6 +677,37 @@ struct IfExprElse {
   AnyExprId else_result;
 };
 
+// Choice nodes
+// ------------
+
+using ChoiceIntroducer = LeafNode<NodeKind::ChoiceIntroducer>;
+
+struct ChoiceSignature {
+  static constexpr auto Kind =
+      NodeKind::ChoiceDefinitionStart.Define(NodeCategory::None);
+
+  ChoiceIntroducerId introducer;
+  AnyNameComponentId name;
```

Comment:
We probably need a `llvm::SmallVector<AnyModifierId> modifiers;` here for things like `public choice X { ... }`.